services/streammanager/googlecloudfunction/googlecloudfunction.go

```diff
@@ -0,0 +1,261 @@
+	}
+
+	opts := []option.ClientOption{
+		option.WithoutAuthentication(),
+	}
+
+	if config.RequireAuthentication {
+		opts = []option.ClientOption{
+			option.WithCredentialsJSON([]byte(config.Credentials)),
+		}
+	}
+
+	service, err := generateService(opts...)
+	// If err is not nil then return
+	if err != nil {
+		pkgLogger.Errorf("[GoogleCloudFunction] error  :: %w", err)
+		return nil, err
+	}
+
+	functionName := getFunctionName(config.GoogleCloudFunctionUrl)
+
+	client := &Client{service, o}
+	destConfig := &Config{config.Credentials, config.FunctionEnvironment, config.RequireAuthentication, config.GoogleCloudFunctionUrl, functionName}
+
+	return &GoogleCloudFunctionProducer{client, destConfig}, err
+}
+
+func (producer *GoogleCloudFunctionProducer) Produce(jsonData json.RawMessage, _ interface{}) (statusCode int, respStatus, responseMessage string) {
+	destConfig := producer.config
+	parsedJSON := gjson.ParseBytes(jsonData)
+
+	if destConfig.FunctionEnvironment == "gen1" {
+		return invokeGen1Functions(producer.client, destConfig.FunctionName, parsedJSON)
+	}
+
+	return invokeGen2Functions(destConfig.GoogleCloudFunctionUrl, destConfig.Credentials, destConfig.RequireAuthentication, parsedJSON)
+}
+
+func invokeGen1Functions(client *Client, functionName string, parsedJSON gjson.Result) (statusCode int, respStatus, responseMessage string) {
+	if client == nil {
+		respStatus = "Failure"
+		responseMessage = "[GoogleCloudFunction] error  :: Failed to initialize GoogleCloudFunction client"
+		return 400, respStatus, responseMessage
+	}
+
+	requestPayload := &cloudfunctions.CallFunctionRequest{
+		Data: parsedJSON.String(),
+	}
+
+	// Make the HTTP request
+	call := client.service.Projects.Locations.Functions.Call(functionName, requestPayload)
+
+	response, err := call.Do()
+
+	if err != nil {
+		statCode, serviceMessage := handleServiceError(err)
+		respStatus = "Failure"
+		responseMessage = "[GOOGLE_CLOUD_FUNCTION] error :: Function call was not executed (Not Modified :: " + serviceMessage
+		pkgLogger.Errorf("[GOOGLE_CLOUD_FUNCTION] error while calling the Gen1 function :: %v", err)
+		return statCode, respStatus, responseMessage
+	}
+
+	// ---------- To be Removed
+	// Process the response (sample response handling here).
+	if response != nil {
+		fmt.Printf("Function call status code: %d\n", response.HTTPStatusCode)
+		// Handle response content as needed.
+	}
+	fmt.Println("Request successful!")
+	// -----------
+
+	respStatus = "Success"
+	responseMessage = "[GoogleCloudFunction] :: Message Payload inserted with messageId :: " + parsedJSON.Get("id").String()
+	return 200, respStatus, responseMessage
+}
+
+func invokeGen2Functions(functionUrl, credentials string, requireAuthentication bool, parsedJSON gjson.Result) (statusCode int, respStatus, responseMessage string) {
+	ctx := context.Background()
+
+	jsonBytes := []byte(parsedJSON.String())
```

Comment:
I think we can get bytes directly from `gjson.Result`. Did you already check it ?